pxd.c

```diff
@@ -1580,29 +1580,32 @@ ssize_t pxd_export(struct fuse_conn *fc, uint64_t dev_id)
 		goto cleanup;
 	}
 
+	spin_unlock(&pxd_dev->lock);
```

Comment:
Couple of issues here. 
1. We set the pxd->exported with the spinlock held, but is checked without grabbing the lock. This is existing code, not your changes.
2. With this change the calls like  pxd_bus_add_dev, device_add_disk can happen in parallel for the same device, which previously didnt happen.

I recommend below approaches to avoid the sleep with the spinlock held:
1. Consider a state in the pxd_dev which indicates that the device is undergoing the process of exporting. Iff the current exported bool is not enough.
The code will now look as below:
```
pxd_export() {
     spin_lock(&pxd_dev->lock);
     if (pxd_dev->exporting) {
           return -EBUSY;
     }
     pxd_dev->exporting = true;
     ...
     spin_unlock(&pxd_dev->lock);
     err = pxd_bus_add_dev(pxd_dev);
     err = device_add_disk(&pxd_dev->dev, pxd_dev->disk, NULL);
     spin_lock(&pxd_dev->lock);
     pxd_dev->exported = true;
     spin_unlock(&pxd_dev->lock);
     ...
}
```
Use atomic if it makes more sense here.